tests/libnode/node.go

```diff
@@ -120,25 +122,31 @@ func GetNodeDrainKey() string {
 	return virtconfig.NodeDrainTaintDefaultKey
 }
 
-func AddLabelToNode(nodeName string, key string, value string) {
+func addLabelAnnotationHelper(nodeName, key, value string, isLabel bool) {
 	virtCli, err := kubecli.GetKubevirtClient()
 	util.PanicOnError(err)
-	node, err := virtCli.CoreV1().Nodes().Get(context.Background(), nodeName, k8smetav1.GetOptions{})
-	Expect(err).ToNot(HaveOccurred())
 
-	old, err := json.Marshal(node)
+	origNode, err := virtCli.CoreV1().Nodes().Get(context.Background(), nodeName, k8smetav1.GetOptions{})
 	Expect(err).ToNot(HaveOccurred())
-	new := node.DeepCopy()
-	new.Labels[key] = value
 
-	newJson, err := json.Marshal(new)
-	Expect(err).ToNot(HaveOccurred())
+	newNode := origNode.DeepCopy()
+	if isLabel {
+		newNode.Labels[key] = value
+	} else {
+		newNode.Annotations[key] = value
+	}
 
-	patch, err := strategicpatch.CreateTwoWayMergePatch(old, newJson, node)
-	Expect(err).ToNot(HaveOccurred())
+	// This is done in an inefficient way since we can patch only labels/annotations here.
+	err = nodes.PatchNode(virtCli, origNode, newNode)
+	Expect(err).ShouldNot(HaveOccurred())
+}
 
-	_, err = virtCli.CoreV1().Nodes().Patch(context.Background(), node.Name, types.StrategicMergePatchType, patch, k8smetav1.PatchOptions{})
-	Expect(err).ToNot(HaveOccurred())
+func AddLabelToNode(nodeName, key, value string) {
+	addLabelAnnotationHelper(nodeName, key, value, true)
+}
+
+func AddAnnotationToNode(nodeName, key, value string) {
+	addLabelAnnotationHelper(nodeName, key, value, false)
```

Comment:
Hey ,
I'm  confused are we adding annotation or label?

Reply:
There are two functions, `AddLabelToNode()` to add a label and `AddAnnotationToNode()` to add an annotation.
Since the implementation is almost identical, they both call a `addLabelAnnotationHelper()` helper function with a bool argument to state whether the desired key/value should be added as a label or annotation.